lib/posthog/exception_capture.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+# Portions of this file are derived from getsentry/sentry-ruby by Software, Inc. dba Sentry
+# Licensed under the MIT License
+# - sentry-ruby/lib/sentry/interfaces/single_exception.rb
+# - sentry-ruby/lib/sentry/interfaces/stacktrace_builder.rb
+# - sentry-ruby/lib/sentry/backtrace.rb
+# - sentry-ruby/lib/sentry/interfaces/stacktrace.rb
+# - sentry-ruby/lib/sentry/linecache.rb
+
+# 💖 open source (under MIT License)
+
+module PostHog
+  module ExceptionCapture
+    RUBY_INPUT_FORMAT = /
+        ^ \s* (?: [a-zA-Z]: | uri:classloader: )? ([^:]+ | <.*>):
+        (\d+)
+        (?: :in\s('|`)(?:([\w:]+)\#)?([^']+)')?$
+      /x
+
+    def self.build_exception_properties(exception)
+      exception_info = build_single_exception(exception)
+
+      {
+        '$exception_list' => [exception_info]
+      }
+    end
+
+    def self.build_single_exception(exception)
+      {
+        'type' => exception.class.to_s,
+        'value' => exception.message || '',
```

Comment:
Ruby is untyped... what happens in the case when a user passes a string rather than an Exception to the `capture_exception` method. I'm guessing this will throw